uni/lib/controller/fetchers/course_units_fetcher/course_units_info_fetcher.dart

```diff
@@ -26,6 +28,32 @@ class CourseUnitsInfoFetcher implements SessionDependantFetcher {
     return parseCourseUnitSheet(response);
   }
 
+  Future<List<Map<String, List<CourseUnitFile>>>> fetchCourseUnitFiles(
```

Comment:
As I understand, the map is a "list" of folders indexed by id and the wrapping list is an artefact of the fact that there are multiple faculties. Is this the case?
If so, I would suggest a structure such as `List<CourseUnitFileDirectory>` where `CourseUnitFileDirectory` is a model that contains a name and a list of files.